packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -447,26 +554,44 @@ export class JobDefinition extends Resource implements IJobDefinition {
           ? this.buildLogConfigurationSecretOptions(container.logConfiguration.secretOptions)
           : undefined,
       } : undefined,
-      memory: container.memoryLimitMiB || 4,
       mountPoints: container.mountPoints,
       privileged: container.privileged || false,
-      resourceRequirements: container.gpuCount
-        ? [{ type: 'GPU', value: String(container.gpuCount) }]
-        : undefined,
+      networkConfiguration: container.networkConfiguration,
       readonlyRootFilesystem: container.readOnly || false,
       ulimits: container.ulimits,
       user: container.user,
-      vcpus: container.vcpus || 1,
       volumes: container.volumes,
+      fargatePlatformConfiguration: container.fargatePlatformConfiguration ? {
+        platformVersion: container.fargatePlatformConfiguration.platformVersion,
+      } : undefined,
     };
+
+    if (!isFargate) {
+      Object.assign(containerProperties, {
```

Comment:
Same - just compute before needing to use it, and declare the types it uses so its easy to navigate and deep dive.

Reply:
Done